### PR TITLE
Dynamic: Improved threat logging, no triple refunds

### PR DESCRIPTION
**Pull request by kriskog**
## About The Pull Request
Corrects an issue with the new midround ruleset selection and cleanup process causing them to not spend threat and refund twice if they failed to execute.
Improved, corrected and added more threat logging where needed.

## Why It's Good For The Game
Bug fixes

## Changelog
:cl: Skoglol
fix: Some dynamic mode threat spending/refunding and threat logging fixes.
fix: Fixed a bug causing some midround antag ghost rolls to fail. Dynamic midround nightmare should now spawn again.
/:cl:

### CODE REVIEW

File: code/game/gamemodes/dynamic/dynamic_rulesets_midround.dm [hunk 1]
```diff
@@ -98,12 +98,10 @@
 	log_game("DYNAMIC: Polling [possible_volunteers.len] players to apply for the [name] ruleset.")
 
 	candidates = pollGhostCandidates("The mode is looking for volunteers to become [antag_flag] for [name]", antag_flag, SSticker.mode, antag_flag, poll_time = 300)
```
Comment (TheChosenEvilOne) [inline code/game/gamemodes/dynamic/dynamic_rulesets_midround.dm]: You might as well fix this while at it:
```suggestion
	candidates = pollGhostCandidates("The mode is looking for volunteers to become [antag_flag] for [name]", antag_flag, SSticker.mode, antag_flag_override ? antag_flag_override : antag_flag, poll_time = 300)
```

Comment (kriskog) [reply code/game/gamemodes/dynamic/dynamic_rulesets_midround.dm]: Sure

File: code/game/gamemodes/dynamic/dynamic.dm [hunk 2]
```diff
@@ -585,10 +587,10 @@ GLOBAL_VAR_INIT(dynamic_forced_threat_level, -1)
 /// Mainly here to facilitate delayed rulesets. All midround/latejoin rulesets are executed with a timered callback to this proc.
 /datum/game_mode/dynamic/proc/execute_midround_latejoin_rule(sent_rule)
 	var/datum/dynamic_ruleset/rule = sent_rule
+	spend_threat(rule.cost)
+	threat_log += "[worldtime2text()]: [rule.ruletype] [rule.name] spent [rule.cost]"
```
Comment (ExcessiveUseOfCobblestone) [inline code/game/gamemodes/dynamic/dynamic.dm]: If the rule fails to execute this will still spend the threat.

Comment (kriskog) [reply code/game/gamemodes/dynamic/dynamic.dm]: It runs clean_up (with built in refunding) on line 607, which makes for better logging and threat management overall. If someone does something stupid later like make the execute delayed, this is also a little more robust in the sense that noone else can spend our threat. Midrounds already relied on spending happening before the final execute which might fail due to no candidates, so this was the intended flow from the start.

**Pull request merged**